test/generateTasks.spec.js

```diff
@@ -52,27 +53,57 @@ describe('generateTasks', () => {
           '*.js': 'lint'
         }
       },
-      {
-        'test.js': '/root/test.js'
-      }
+      ['test.js']
     )
     const commands = result.map(match => match.commands)
     expect(commands).toEqual(['lint'])
   })
 
+  it('should return absolute paths', () => {
+    const result = generateTasks(
+      {
+        gitDir: '..',
+        linters: {
+          '*': 'lint'
+        }
+      },
+      files
+    )
+    result[0].fileList.forEach(file => {
+      expect(path.isAbsolute(file)).toBe(true)
+    })
+  })
+
   it('should return an empty file list for linters with no matches.', () => {
-    const result = generateTasks(linters, files)
+    const result = generateTasks(config, files)
     result.forEach(task => {
       if (task.commands === 'unknown-js') {
         expect(task.fileList.length).toEqual(0)
       } else {
-        expect(task.fileList.length).toNotEqual(0)
+        expect(task.fileList.length).not.toEqual(0)
       }
     })
   })
 
+  it('should match pattern "*.js" for relative path', () => {
+    const relPath = path.resolve(path.join(process.cwd(), '..'))
+    const result = generateTasks(Object.assign({}, config, { gitDir: '..' }), files)
+    const linter = result.find(item => item.pattern === '*.js')
+    expect(linter).toEqual({
+      pattern: '*.js',
+      commands: 'root-js',
+      fileList: [
```

Comment:
I think `path.normalize` can help

Reply:
Since I don't have windows, would you mind trying to fix it after this is merged? This should not affect Windows users but it would be great if all contributors could see green tests locally.

Reply:
Sure, no problem.

Reply:
If you or @sudo-suhas have time for that it would be definitively something beneficial. Do I need to signup for it?

Reply:
yeah I can make a PR to add it.

> Do I need to signup for it?
Let me set the config first in new PR than we can discuss it there

Reply:
Here's an example cfg file - https://github.com/kentcdodds/cross-env/blob/master/appveyor.yml